citrination_client/client.py

```diff
@@ -517,12 +593,18 @@ def _put_with_version_check(self, url, data, headers):
         result = requests.put(url, data=data, headers=headers, proxies=self.proxies)
         return self._check_response_for_errors(result)
 
+    def _delete_with_version_check(self, url, headers):
+        result = requests.delete(url, headers=headers, proxies=self.proxies)
+        return self._check_response_for_errors(result)
+
     def _check_search_response_for_errors(self, response):
         if response.status_code == 204 or response.status_code == 524:
             raise RequestTimeoutException()
         return self._check_response_for_errors(response)
 
     def _check_response_for_errors(self, response):
+        if response.status_code > 399:
```

Comment:
Doesn't this supersede the `if response.status_code == 400` later in the function?